src/util/customErrors.ts

```diff
@@ -0,0 +1,26 @@
+/**
+ * A custom error that holds additional information and a more explicit message
+ * to simplify tracking down errors that occur during execution
+ */
+export class AnnotatedError extends Error {
+  // TODO: need to explicitly specify cause in TypeScript versions < 4.7
+  // Remove this when TypeScript is upgraded, and add the cause directly to the `super` call
+  cause: Error;
+
+  constructor(
+    originalError: Error,
+    message: string,
+    public expressionName: string,
```

Comment:
Wait, this `public name` make them instance properties without having to re-iterate yourself with `this.name = name` on all the things?

Reply:
😎